mountpoint-s3-client/src/mock_client.rs

```diff
@@ -734,6 +733,48 @@ mod tests {
         check_continuation!("/", 2, "dirs/dir2/", &keys[7..9], &[]);
     }
 
+    #[tokio::test]
+    async fn list_objects_unicode() {
+        let client = MockClient::new(MockClientConfig {
+            bucket: "test_bucket".to_string(),
+            part_size: 1024,
+        });
+
+        let mut keys = vec![];
+        for i in 0..5 {
+            keys.push(format!("dirs/こんにちは/file{i}.txt"));
+        }
+        for i in 0..5 {
+            keys.push(format!("dirs/😄🥹/file{i}.txt"));
```

Comment:
We probably want to test with negative case here? for example, if we change the second prefix to `dirs/😄🥹😮` the test would panick without your code change.

Reply:
Yep, good call, added a couple of negative tests.